app/src/main/java/me/saket/dank/widgets/IndentedLayout.java

```diff
@@ -54,18 +98,30 @@ public IndentedLayout(Context context, @Nullable AttributeSet attrs) {
   protected void onLayout(boolean changed, int l, int t, int r, int b) {
     super.onLayout(changed, l, t, r, b);
 
-    indentationLinePath.reset();
+    trees = new ArrayList<ColoredTree>();
+    Path linePath;
+
     for (int i = 0; i < indentationDepth; i++) {
       float lineStartX = spacePerDepthPx * (i + 1) + indentationLinePaint.getStrokeWidth();
-      indentationLinePath.moveTo(lineStartX, 0);
-      indentationLinePath.lineTo(lineStartX, getHeight());
+
+      linePath = new Path();
+      linePath.moveTo(lineStartX, 0);
+      linePath.lineTo(lineStartX, getHeight());
+
+      trees.add(new ColoredTree((this.coloredDepth == true ? Color.parseColor(INDENTATION_COLORS[i]) : Color.LTGRAY), linePath));
     }
   }
 
   @Override
   public void draw(Canvas canvas) {
     super.draw(canvas);
-    canvas.drawPath(indentationLinePath, indentationLinePaint);
+
+    for(ColoredTree tree : trees) {
+      indentationLinePaint.setColor(tree.getColor());
+      canvas.drawPath(tree.getPath(), indentationLinePaint);
+    }
+
+    invalidate();
```

Comment:
Is this call to `invalidate` necessary? It's done inside of `draw` method which is a bit suspicious